boards/BMS/Src/App/states/App_BalanceState.c

```diff
@@ -0,0 +1,67 @@
+#include <stdlib.h>
+#include "states/App_InitState.h"
+#include "states/App_BalanceState.h"
+#include "states/App_AllStates.h"
+
+static void BalanceStateRunOnEntry(struct StateMachine *const state_machine)
+{
+    struct BmsWorld *         world  = App_SharedStateMachine_GetWorld(state_machine);
+    struct BmsCanTxInterface *can_tx = App_BmsWorld_GetCanTx(world);
+    struct Airs *             airs   = App_BmsWorld_GetAirs(world);
+
+    App_CanTx_SetPeriodicSignal_STATE(can_tx, CANMSGS_BMS_STATE_MACHINE_STATE_BALANCE_CHOICE);
+    App_Airs_OpenAirPositive(airs);
+}
+
+static void BalanceStateRunOnTick1Hz(struct StateMachine *const state_machine)
+{
+    App_AllStatesRunOnTick1Hz(state_machine);
+}
+
+static void BalanceStateRunOnTick100Hz(struct StateMachine *const state_machine)
+{
+    struct BmsWorld *         world       = App_SharedStateMachine_GetWorld(state_machine);
+    struct Accumulator *      accumulator = App_BmsWorld_GetAccumulator(world);
+    struct BmsCanRxInterface *can_rx      = App_BmsWorld_GetCanRx(world);
+    struct BmsCanTxInterface *can_tx      = App_BmsWorld_GetCanTx(world);
+
+    if (App_AllStatesRunOnTick100Hz(state_machine))
+    {
+        const struct State *next_state           = App_GetBalanceState();
+        bool                is_charging_complete = false;
+        bool                is_balancing_enabled = true;
+
+        if (App_Accumulator_IsPackBalanced(accumulator) && App_Accumulator_HasCellReachedMaxVoltageTarget(accumulator))
+        {
+            is_charging_complete = true;
+            is_balancing_enabled = false;
+            next_state           = App_GetInitState();
+        }
+        else if (App_CanRx_CAN_DEBUGGING_SIGNALS_GetSignal_DISABLE_CELL_BALANCING(can_rx))
+        {
+            // Overriding cell balancing via CAN, transition back into the init state
+            is_balancing_enabled = false;
+            next_state           = App_GetInitState();
+        }
+
+        App_Accumulator_SetIsCellBalancingEnabled(accumulator, is_balancing_enabled);
+
+        // Update is charging complete status on CAN
+        App_CanTx_SetPeriodicSignal_IS_CHARGING_COMPLETE(can_tx, is_charging_complete);
+
+        App_SharedStateMachine_SetNextState(state_machine, next_state);
```

Comment:
Why is this here? shouldn't we just wait in balancing state until the cells are within the specified range?

Reply:
next state will only change if we run into the if, and else if statements above. notice how next_state is initialized to App_GetBalanceState()